### PR TITLE
update for tetris

### PR DESCRIPTION
**Pull request by YCF**
Add a fake row to config the encoder's keycode easily.
Auto change the rgblight color with the keycode.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tetris/config.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/tetris/config.h]: Please revert these changes.

Specifically, we want `#pragma once` here.

File: keyboards/tetris/config.h [hunk 2]
```diff
@@ -36,22 +32,30 @@
 /* Locking resynchronize hack */
 #define LOCKING_RESYNC_ENABLE
 
-#define NO_ACTION_MACRO
-#define NO_ACTION_FUNCTION
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (drashna) [inline keyboards/tetris/config.h]: Please remove this, as it's the default, and already set.

Additionally, please re-add the `NO_ACTION` defines.

File: keyboards/tetris/config.h [hunk 3]
```diff
@@ -36,22 +32,30 @@
...
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
 
-#define TAPPING_TERM 200
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/tetris/config.h]: Please remove this, as it's no longer valid, as it is the default behavior.

File: keyboards/tetris/keymaps/default/keymap.c [hunk 4]
```diff
@@ -1,829 +1,285 @@
...
-
-/* Tap Dance Definitions */
-qk_tap_dance_action_t tap_dance_actions[] = {
-  [TD_Mute_Next]  = ACTION_TAP_DANCE_DOUBLE( KC_MUTE, KC_MNXT )
-  ,[TD_SFT]  = ACTION_TAP_DANCE_FN_ADVANCED( NULL, dance_cln_finished, dance_cln_reset )
-  ,[TD_LBRC]  = ACTION_TAP_DANCE_DOUBLE( KC_LBRC,   KC_LCBR )
-  ,[TD_RBRC]  = ACTION_TAP_DANCE_DOUBLE( KC_RBRC,   KC_RCBR )
-};
-
+#include "tetris.h"
```
Comment (drashna) [inline keyboards/tetris/keymaps/default/keymap.c]: Please revert this to `#include QMK_KEYBOARD_H`

File: keyboards/tetris/keymaps/default/keymap.c [hunk 5]
```diff
@@ -1,829 +1,285 @@
...
-  KC_CAPS, _______, _______, _______, _______, TO(_GAME), _______,   _______, _______, _______, _______, _______,
-  _______, _______, _______, _______, _______, _______,   TO(_NUMB), MU_TOG,  _______, _______, _______, _______,
-  _______, _______, _______, _______, _______,      _______,         _______, _______, _______, _______, _______ ),
-
-[_GAME] = LAYOUT_planck_mit(
-  XXXXXXX,KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    XXXXXXX,
-  XXXXXXX,KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
-  XXXXXXX,KC_LCTL, KC_A,    KC_S,    KC_D,    KC_B,    KC_N,    KC_M,    XXXXXXX, XXXXXXX, KC_UP,   TO( 0 ),
-  XXXXXXX,KC_LSFT, KC_Z,    KC_X,    KC_C,       KC_SPC,        XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN, KC_RGHT ),
+  KEYMAP(
```
Comment (drashna) [inline keyboards/tetris/keymaps/default/keymap.c]: Please revert this:
```suggestion
  [_BASE] = LAYOUT_ortho_5x12(
```

As well as the rest

File: keyboards/tetris/keymaps/default/keymap.c [hunk 6]
```diff
@@ -1,829 +1,285 @@
...
-      RGB_encoder_bool=true;
-      RGB_encoder_count=5;
-      rgblight_mode(1);
-      rgblight_setrgb( 0,0,0 );
-    }else if (RGB_encoder_bool && timer_elapsed(RGB_encoder_timer) < 800) {
-      RGB_encoder_timer=timer_read();
-      RGB_encoder_count-=1;
-      if (RGB_encoder_count>5 || RGB_encoder_count<1){
-        RGB_encoder_count=5;
+  if (RGB_encoder_dir != 0) {
```
Comment (drashna) [inline keyboards/tetris/keymaps/default/keymap.c]: If this is set to the default color, then you don't need this anymore. 

You can use `keyboard_post_init_user` instead, as this runs after the rgblight feature has been initialized.

File: keyboards/tetris/rules.mk [hunk 7]
```diff
@@ -36,26 +36,23 @@ F_USB = $(F_CPU)
 # Interrupt driven control endpoint task(+60)
 OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
 
+
 # Boot Section Size in *bytes*
-#OPT_DEFS += -DBOOTLOADER_SIZE=4096
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/tetris/rules.mk]: Please revert the changes to this file. 

Namely, 

`BOOTLOADER = qmk-dfu` sets the bootloader size, so that's not needed. 

And please remove all of the question marks.

And add `LAYOUTS = LAYOUT_ortho_5x12`

File: keyboards/tetris/tetris.h [hunk 8]
```diff
@@ -3,16 +3,18 @@
 
 #include "quantum.h"
 
-#define LAYOUT_planck_mit( \
+#define KEYMAP( \
```
Comment (drashna) [inline keyboards/tetris/tetris.h]: Please revert this change.

```suggestion
#define LAYOUT_ortho_5x12( \	
```

File: keyboards/tetris/keymaps/default/keymap.c [hunk 9]
```diff
@@ -1,829 +1,285 @@
...
+    KC_VOLD, KC_VOLU, KC_PGDN, KC_PGUP),
+
+  KEYMAP(
+    KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_BSPC, KC_TRNS, KC_DEL, KC_TRNS, KC_TRNS, KC_BSLS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+  KEYMAP(
```
Comment (drashna) [inline keyboards/tetris/keymaps/default/keymap.c]: Please remove all of the empty layers.

File: keyboards/tetris/keymaps/default/keymap.c (hunk 5 above)
Comment (noroadsleft) [reply keyboards/tetris/keymaps/default/keymap.c]: This should actually be `LAYOUT_planck_mit` rather than `LAYOUT_ortho_5x12`.

File: keyboards/tetris/tetris.h (hunk 8 above)
Comment (noroadsleft) [reply keyboards/tetris/tetris.h]: Should be `LAYOUT_planck_mit`.

```suggestion
#define LAYOUT_planck_mit( \
```

File: keyboards/tetris/keymaps/default/keymap.c (hunk 6 above)
Comment (YCF) [reply keyboards/tetris/keymaps/default/keymap.c]: Some rgblight  animation will show up with the encoder,this codes is the animation code.

File: keyboards/tetris/config.h [hunk 10]
```diff
@@ -1,4 +1,6 @@
 #pragma once
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/tetris/config.h]:
```suggestion
```

File: keyboards/tetris/config.h [hunk 11]
```diff
@@ -1,4 +1,6 @@
...
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/tetris/config.h]:
```suggestion
```

File: keyboards/tetris/config.h [hunk 12]
```diff
@@ -39,19 +36,22 @@
...
 #define RGBLIGHT_ANIMATIONS
-#define RGBLIGHT_SLEEP
 #define RGBLED_NUM 47
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
 #define RGBLIGHT_EFFECT_KNIGHT_LED_NUM 12
-//#define RGBLIGHT_LIMIT_VAL 128
+
 #endif
```
Comment (drashna) [inline keyboards/tetris/config.h]:
```suggestion
```

File: keyboards/tetris/rules.mk [hunk 13]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
-NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
-BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
-AUDIO_ENABLE = no
-RGBLIGHT_ENABLE = yes
-TAP_DANCE_ENABLE = no
-
-EXTRAFLAGS += -flto # Make the hex smaller
-
-LAYOUTS = planck_mit
+BOOTMAGIC_ENABLE ?= no	# Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no	# Virtual DIP switch configuration(+1000)
```

File: keyboards/tetris/rules.mk [hunk 14]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
```

File: keyboards/tetris/rules.mk [hunk 15]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
```

File: keyboards/tetris/rules.mk [hunk 16]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
CONSOLE_ENABLE = no	# Console for debug(+400)
```

File: keyboards/tetris/rules.mk [hunk 17]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
COMMAND_ENABLE = no    # Commands for debug and configuration
```

File: keyboards/tetris/rules.mk [hunk 18]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```

File: keyboards/tetris/rules.mk [hunk 19]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/tetris/rules.mk [hunk 20]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```

File: keyboards/tetris/rules.mk [hunk 21]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+AUDIO_ENABLE ?= yes
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
AUDIO_ENABLE = yes
```

File: keyboards/tetris/rules.mk [hunk 22]
```diff
@@ -36,26 +36,25 @@ F_USB = $(F_CPU)
...
+RGBLIGHT_ENABLE ?= yes
```
Comment (drashna) [inline keyboards/tetris/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = yes
```

**Comment by YCF** [comment]
thanks @drashna

**Comment by drashna** [comment]
it looks like there is an irreconcilably merge conflict.
